js/style/parse_color.js

```diff
@@ -22,9 +22,14 @@ function parseColor(input) {
 
     // Color string
     } else if (typeof input === 'string') {
-        var output = colorDowngrade(parseCSSColor(input));
-        colorCache[input] = output;
-        return output;
+        var parsedColor = parseCSSColor(input);
+        if (parsedColor !== null) {
```

Comment:
Use a guard clause (`if (!parsedColor) { throw new Error('Invalid color ' + input); }`) and unindent the happy code path.